### PR TITLE
Add network policies

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/4033

### CODE REVIEW

File: helm/kubernetes-metrics-server-chart/templates/np.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    matchLabels:
+      app: {{ .Values.name }}
+  ingress:
+  - ports:
+    - port: {{ .Values.port }}
+      protocol: TCP
+  egress:
+  - to:
+    - ipBlock:
+        cidr: 10.0.0.0/8
```
Comment (pipo02mix) [inline helm/kubernetes-metrics-server-chart/templates/np.yaml:20]: Same question as I did for ksm, do we need to change for node IP ranges which are not the default ones?

**Review by pipo02mix — COMMENTED** [review]
Question asked

File: helm/kubernetes-metrics-server-chart/templates/np.yaml (hunk 1 above)
Comment (corest) [reply helm/kubernetes-metrics-server-chart/templates/np.yaml:20]: Same answer - this allows all the private ips. 

So, those services which needs access to some external IPs has egress policy which allows all egress access (as there is no way to setup dns based restriction)

And those which do not need externall access - have egress policy only for accessing internal ips (that covers any host network ip within control-plane and tenant )

**Review by calvix — COMMENTED** [review]
lgtm, i assume its tested

**Review by pipo02mix — APPROVED** [review]
LGTMIIWFY

**Pull request merged**